scapy/modules/p0f.py

```diff
@@ -459,8 +459,8 @@ def p0f_impersonate(pkt, osgenre=None, osdetails=None, signature=None,
         pkt.payload.window = mtu * int(pers[0][1:])
     elif pers[0][0] == 'S':
         ## needs MSS set
-        MSS = filter(lambda x: x[0] == 'MSS', options)
-        if not filter(lambda x: x[0] == 'MSS', options):
+        MSS = [x for x in options if x[0] == 'MSS']
+        if not [x for x in options if x[0] == 'MSS']:
```

Comment:
I think `mss` would be a better variable name than `MSS`. Also, I'm not sure to understand but I think this should work and be more efficient:
```Python
mss = [x for x in options if x[0] == 'MSS']
if not mss:
```